### PR TITLE
Added support for C# using OmniSharp

### PR DESCRIPTION
**Pull request by bjarkevad**
Added initial support for C#, comments et al is of course very welcome.
OmniSharp seems to work reasonably well for the tests I've done on Linux with Mono, but someone should probably try it out with some larger solutions on other system configs.

### CODE REVIEW

File: contrib/lang/csharp/packages.el [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+              "mgs"   'omnisharp-helm-find-symbols
+              "mgi"   'omnisharp-find-implementations
+              "mgr"   'omnisharp-navigate-to-region
+              "mgm"   'omnisharp-navigate-to-solution-member
+              "mgM"   'omnisharp-navigate-to-solution-member-other-window
+              "mgf"   'omnisharp-navigate-to-solution-file
+              "mgF"   'omnisharp-navigate-to-solution-file-then-file-member
+
+              ;; Type info
+              "mt" 'omnisharp-current-type-information
```
Comment (syl20bnr) [inline contrib/lang/csharp/packages.el]: you have a clash here with the test prefix `mt`

File: contrib/lang/csharp/packages.el [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+(defvar csharp-excluded-packages '()
+  "List of packages to exclude.")
+
+(defun csharp/init-omnisharp ()
+  ;; Load omnisharp-mode with csharp-mode, this should start the omnisharp server automatically
+  (add-hook 'csharp-mode-hook 'omnisharp-mode)
+  (use-package omnisharp
+    :defer t
+    :config  (evil-leader/set-key-for-mode 'csharp-mode
```
Comment (syl20bnr) [inline contrib/lang/csharp/packages.el]: Sort the key bindings

File: contrib/lang/csharp/packages.el [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+              ;; Solution/project manipulation
+              "mpa" 'omnisharp-add-to-solution-current-file
+              "mpA" 'omnisharp-add-to-solution-dired-selected-files
+              "mpr" 'omnisharp-remove-from-project-current-file
+              "mpR" 'omnisharp-remove-from-project-dired-selected-files
+              "mpl" 'omnisharp-add-reference
+
+              ;; Compile
+              "mb" 'omnisharp-build-in-emacs ;; Only one compile command so use top-level
```
Comment (syl20bnr) [inline contrib/lang/csharp/packages.el]: should be `mcc` if I'm not mistaken

File: contrib/lang/csharp/packages.el [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+              ;; Code manipulation
+              "mo" 'omnisharp-auto-complete-overrides
```
Comment (syl20bnr) [inline contrib/lang/csharp/packages.el]: not in the conventions but we can keep `mo` for the user as well as `o`

File: contrib/lang/csharp/packages.el (hunk 2 above)
Comment (bjarkevad) [inline contrib/lang/csharp/packages.el]: How do you want the sorting to be?

**Comment by bjarkevad** [comment]
I've updated the bindings, but I'm still not really satisfied with the mnemonics - suggestions?

**Comment by bjarkevad** [comment]
Updated bindings again :)

**Comment by syl20bnr** [comment]
Very nice layer ! :heart: 
Thank you !

Cherry-picked into `develop` you can safely delete your branch.

I made some modifications/improvements:
- updated the readme in 8e4eb673ee9561149ecbb8b095eb851b7d82c071
- sorted the key bindings in c5aa369f675e37fc32068aa0e0f259d6485a3132
- change logo in 68a004c892a9ab13818f87e354f56ea6d64fc277

**Pull request closed without merging**